app/graph.py

```diff
@@ -471,23 +489,119 @@ def update(self):
+                            'node_pod_sec_iface_conn': [],
+                            'node_leaf_ter_iface_conn': [],
+                            'node_pod_ter_iface_conn': [],
+                            'node_leaf_all_iface_conn': [],
+                        }
+
+                    pods = nodes[node_name]['pods']
+                    pod_name = i.metadata.name
+                    pods[pod_name] = {
+                        "ip": i.status.pod_ip,
+                        "primary_iface": "",
+                        "ns": i.metadata.namespace,
+                        "labels": i.metadata.labels if i.metadata.labels is not None else {},
+                        "other_ifaces": {},
+                        "annotations": i.metadata.annotations if i.metadata.annotations is not None else {},
                     }
 
-                self.topology['nodes'][i.spec.node_name]['pods'][i.metadata.name] = {
-                    "ip": i.status.pod_ip,
-                    "ns": i.metadata.namespace,
-                    "labels": i.metadata.labels if i.metadata.labels is not None else {}
-                    }
-                    
+                    annotations = i.metadata.annotations
+                    if annotations is not None:
+                        for key, annotation in annotations.items():
+                            if key == "k8s.v1.cni.cncf.io/network-status":
+                                items_list = json.loads(annotation)
+                                pod_ip = i.status.pod_ip
+                                for val in items_list:
+                                    if val["ips"][0] != pod_ip:
+                                        iface_name = str(val["name"].split('/')[-1])
+                                        pod_iface = str(val["interface"])
+                                        pods[pod_name]['other_ifaces'][iface_name] = pod_iface
+                                    else:
+                                        pods[pod_name]['primary_iface'] = str(val["interface"])
+            except Exception as e:
+                logger.error(f"Error processing pod: {i.metadata.name}. Error: {str(e)}")
+
+        try:
+            cr = self.custom_obj.list_namespaced_custom_object(group="aci.fabricattachment", version="v1", namespace="aci-containers-system", plural="nodefabricnetworkattachments")
+
+            if cr.get("items"):
+                for nodeName in self.topology['nodes']:
+                    try:
+                        for i in cr.get("items"):
+                            aciTopology = i["spec"].get("aciTopology")
+                            if aciTopology is not None and i["spec"]["nodeName"] == nodeName:
+                                for iface_name, link in i["spec"]["aciTopology"].items():
+                                    pods = link.get("pods", [])
+                                    if pods:
+                                        if "sriov" in i["metadata"]["name"]:
```

Comment:
All instances of `if "sriov" in i["metadata"]["name"]:` should be replaced with `if "sriov" in i["spec"]["primaryCni"]:`
there is a specific field (`primaryCni`) in the `nodefabricnetworkattachments` that can tell us which `primaryCni` we are using and we should not have to rely on the `nodefabricnetworkattachments` name.